### PR TITLE
Superassign

### PR DESCRIPTION
**Pull request by JanJecmen**

### CODE REVIEW

File: rir/src/interpreter/interp.c [hunk 1]
```diff
@@ -1216,6 +1238,17 @@ INSTRUCTION(stvar_) {
...
 
+INSTRUCTION(stvar2_) {
+    SEXP sym = readConst(ctx, pc);
+    int wasChanged = FRAME_CHANGED(env);
+    SLOWASSERT(TYPEOF(sym) == SYMSXP);
+    SEXP val = ostack_pop(ctx);
+    INCREMENT_NAMED(val);
+    setVar(sym, val, ENCLOS(env));
+    if (!wasChanged)
+        CLEAR_FRAME_CHANGED(env);
```
Comment (o-) [inline rir/src/interpreter/interp.c]: the FRAME_CHANGED keeps track of whether some non-local change to the env happened. Here this is the case (super assign is non local). Also the change happens on ENCLOS(env). So you can just delete lines 124[3,8,9]

File: rir/src/ir/Compiler.cpp [hunk 2]
```diff
@@ -475,6 +478,11 @@ bool compileSpecialCall(Context& ctx, SEXP ast, SEXP fun, SEXP args_) {
             //            if (getter == symbol::Bracket)
         }
 
+        if (superAssign) {
```
Comment (o-) [inline rir/src/ir/Compiler.cpp:484]: can you move the guard after the condition (only inserting it if needed) instead of removing it afterwards?

**Review by o- — COMMENTED** [review]
thanks, that looks good (except the two nitpicks)

File: rir/src/interpreter/interp.c (hunk 1 above)
Comment (JanJecmen) [reply rir/src/interpreter/interp.c]: Thanks for the explanation. Fixed that.

**Pull request merged**